tests/test_efficientnet.py

```diff
@@ -0,0 +1,229 @@
+
+if TYPE_CHECKING:
+    import PIL
+
+    has_pil = True
+else:
+    PIL, has_pil = optional_import("PIL")
+
+
+def get_model_names():
+    return ["efficientnet-b{}".format(d) for d in range(8)]
+
+
+def get_expected_model_shape(model_name):
+    model_input_shapes = {
+        "efficientnet-b0": 224,
+        "efficientnet-b1": 240,
+        "efficientnet-b2": 260,
+        "efficientnet-b3": 300,
+        "efficientnet-b4": 380,
+        "efficientnet-b5": 456,
+        "efficientnet-b6": 528,
+        "efficientnet-b7": 600,
+    }
+    return model_input_shapes[model_name]
+
+
+def make_shape_cases(models, spatial_dims, batches, pretrained, in_channels=3, num_classes=1000):
+    ret_tests = []
+    for spatial_dim in spatial_dims:  # selected spatial_dims
+        for batch in batches:  # check single batch as well as multiple batch input
+            for model in models:  # selected models
+                for is_pretrained in pretrained:  # pretrained or not pretrained
+                    kwargs = {
+                        "model_name": model,
+                        "pretrained": is_pretrained,
+                        "progress": False,
+                        "spatial_dims": spatial_dim,
+                        "in_channels": in_channels,
+                        "num_classes": num_classes,
+                    }
+                    ret_tests.append(
+                        [
+                            kwargs,
+                            (
+                                batch,
+                                in_channels,
+                            )
+                            + (get_expected_model_shape(model),) * spatial_dim,
+                            (batch, num_classes),
+                        ]
+                    )
+    return ret_tests
+
+
+# create list of selected models to speed up redundant tests
+# only test the models B0, B3, B7
+SEL_MODELS = [get_model_names()[i] for i in [0, 3, 7]]
+
+# pretrained=False cases
+# 1D models are cheap so do test for all models in 1D
+CASES_1D = make_shape_cases(
+    models=get_model_names(), spatial_dims=[1], batches=[1, 4], pretrained=[False], in_channels=3, num_classes=1000
+)
+
+# 2D and 3D models are expensive so use selected models
+CASES_2D = make_shape_cases(
+    models=SEL_MODELS, spatial_dims=[2], batches=[1, 4], pretrained=[False], in_channels=3, num_classes=1000
+)
+CASES_3D = make_shape_cases(
+    models=[SEL_MODELS[0]], spatial_dims=[3], batches=[1], pretrained=[False], in_channels=3, num_classes=1000
+)
+
+# pretrained=True cases
+# tabby kitty test with pretrained model
+# needs 'testing_data/kitty_test.jpg'
+CASES_KITTY_TRAINED = [
+    (
+        {
+            "model_name": "efficientnet-b0",
+            "pretrained": True,
+            "progress": False,
+            "spatial_dims": 2,
+            "in_channels": 3,
+            "num_classes": 1000,
+        },
+        os.path.join(os.path.dirname(__file__), "testing_data/kitty_test.jpg"),
```

Comment:
the path needs to be revised for paltform compatibility. cool image, is it with a CC-by license...
```suggestion
        os.path.join(os.path.dirname(__file__), "testing_data", "kitty_test.jpg"),
```

Reply:
Thank you, I will update the code.
The image comes from reddit: https://www.reddit.com/r/aww/comments/h689o/kitten_with_a_bow_tie_xpost_from_pics/
Not entirely sure about the license. If you want, I can find and update to a CC-by license picture for a cat. Please feel free to suggest if you have any suggestions.

Reply:
thanks, public domain image as testing data would be great

Reply:
I have updated the picture with the following that has CC0 License:
https://commons.wikimedia.org/wiki/File:Tabby_cat_with_blue_eyes-3336579.jpg